backend/game_engine/firestore.py

```diff
@@ -414,3 +415,28 @@ def clear_votes(self) -> None:
         votes = self._client.collection('votes').stream()
         with ThreadPoolExecutor(max_workers=self._thread_pool_size) as executor:
             executor.submit(self._delete_vote_fn, votes)
+
+    def find_matchmaker_games(self, region="US", time_now=datetime.datetime.now()) -> list:
+        # Time comparison is done solely based on the hour
+        games_list = []
+        db = self._client
+        games = db.collection(u'games').stream()
+
+        for game in games:
```

Comment:
We can improve performance here by using a predicate in the query. For example:

games = db.collection(u'games').where('country_code, '=', 'US').where('game_has_started', '=', False)

Check out this [link](https://firebase.google.com/docs/firestore/query-data/queries#simple_queries) for reference.